compiler/rustc_query_impl/src/plumbing.rs

```diff
@@ -253,20 +255,17 @@ macro_rules! define_queries {
         }
 
         mod make_query {
+            use rustc_middle::dep_graph::DepKind;
+            use rustc_query_system::ich::StableHashingContext;
             use super::*;
 
-            // Create an eponymous constructor for each query.
-            $(#[allow(nonstandard_style)] $(#[$attr])*
-            pub fn $name<'tcx>(tcx: QueryCtxt<'tcx>, key: <queries::$name<'tcx> as QueryConfig>::Key) -> QueryStackFrame {
-                opt_remap_env_constness!([$($modifiers)*][key]);
-                let kind = dep_graph::DepKind::$name;
-                let name = stringify!($name);
+            fn create_query_frame<'tcx, K: Copy + Key + for<'a> HashStable<StableHashingContext<'a>>>(tcx: QueryCtxt<'tcx>, do_describe: fn(QueryCtxt<'tcx>, K) -> String, key: K, kind: DepKind, name: &'static str) -> QueryStackFrame {
```

Comment:
Can you move this outside the macro?

Reply:
good idea, done :) had to rearrange a bunch of imports because `define_queries` is expanded in the crate root; always confuses me that those are different. I'll see if I can make that the same in a follow-up PR actually.